packages/cache/src/cache.ts

```diff
@@ -153,6 +203,106 @@ export async function restoreCache(
   return undefined
 }
 
+/**
+ * Restores cache using the new Cache Service
+ *
+ * @param paths a list of file paths to restore from the cache
+ * @param primaryKey an explicit key for restoring the cache
+ * @param restoreKeys an optional ordered list of keys to use for restoring the cache if no cache hit occurred for key
+ * @param downloadOptions cache download options
+ * @param enableCrossOsArchive an optional boolean enabled to restore on windows any cache created on any platform
+ * @returns string returns the key for the cache hit, otherwise returns undefined
+ */
+async function restoreCachev2(
+  paths: string[],
+  primaryKey: string,
+  restoreKeys?: string[],
+  options?: DownloadOptions,
+  enableCrossOsArchive = false
+): Promise<string | undefined> {
+  restoreKeys = restoreKeys || []
+  const keys = [primaryKey, ...restoreKeys]
+
+  core.debug('Resolved Keys:')
+  core.debug(JSON.stringify(keys))
+
+  if (keys.length > 10) {
+    throw new ValidationError(
+      `Key Validation Error: Keys are limited to a maximum of 10.`
+    )
+  }
+  for (const key of keys) {
+    checkKey(key)
+  }
+
+  let archivePath = ''
+  try {
+    const twirpClient = cacheTwirpClient.internalCacheTwirpClient()
+    const backendIds: utils.BackendIds = utils.getBackendIdsFromToken()
+    const compressionMethod = await utils.getCompressionMethod()
+
+    const request: GetCacheEntryDownloadURLRequest = {
+      workflowRunBackendId: backendIds.workflowRunBackendId,
+      workflowJobRunBackendId: backendIds.workflowJobRunBackendId,
+      key: primaryKey,
+      restoreKeys,
+      version: utils.getCacheVersion(
+        paths,
+        compressionMethod,
+        enableCrossOsArchive
+      )
+    }
+
+    const response: GetCacheEntryDownloadURLResponse =
```

Comment:
These Twirp methods are typed, so we don't need to define the type here on `response`

```diff
diff --git a/packages/cache/src/cache.ts b/packages/cache/src/cache.ts
index 7d0cd00..2682b06 100644
--- a/packages/cache/src/cache.ts
+++ b/packages/cache/src/cache.ts
@@ -253,9 +253,7 @@ async function restoreCachev2(
       )
     }
 
-    const response: GetCacheEntryDownloadURLResponse =
-      await twirpClient.GetCacheEntryDownloadURL(request)
-
+    const response = await twirpClient.GetCacheEntryDownloadURL(request)
     if (!response.ok) {
       core.warning(`Cache not found for keys: ${keys.join(', ')}`)
       return undefined
@@ -503,8 +501,7 @@ async function saveCachev2(
       key,
       version
     }
-    const response: CreateCacheEntryResponse =
-      await twirpClient.CreateCacheEntry(request)
+    const response = await twirpClient.CreateCacheEntry(request)
     if (!response.ok) {
       throw new ReserveCacheError(
         `Unable to reserve cache with key ${key}, another job may be creating this cache.`

```